data/data/bootstrap/bootstrap-in-place/files/opt/openshift/bootstrap-in-place/bootstrap-in-place-post-reboot.sh

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+set -euoE pipefail ## -E option will cause functions to inherit trap
+
+echo "Running bootkube bootstrap-in-place post reboot"
+export KUBECONFIG=/etc/kubernetes/bootstrap-secrets/kubeconfig
+
+function wait_for_api {
+  until oc get csr &> /dev/null
+  do
+    echo "Waiting for api ..."
+    sleep 5
+  done
+}
+
+function restart_kubelet {
+  echo "Restarting kubelet"
+  until [ "$(oc get pod -n openshift-kube-apiserver-operator --selector='app=kube-apiserver-operator'  -o jsonpath='{.items[0].status.conditions[?(@.type=="Ready")].status}' | grep -c "True")" -eq 1 ];
+  do
+    echo "Waiting for kube-apiserver-operator ready condition to be True"
+    sleep 10
+  done
+  # daemon-reload is required because /etc/systemd/system/kubelet.service.d/20-nodenet.conf is added after kubelet started
+  systemctl daemon-reload
+  systemctl restart kubelet
+
+  while grep  bootstrap-kube-apiserver /etc/kubernetes/manifests/kube-apiserver-pod.yaml;
+  do
+    echo "Waiting for kube-apiserver to apply the new static pod configuration"
+    sleep 10
+  done
+  systemctl restart kubelet
+}
+
+function approve_csr {
+  echo "Approving csrs ..."
+  until [ "$(oc get nodes --selector='node-role.kubernetes.io/master' -o jsonpath='{.items[0].status.conditions[?(@.type=="Ready")].status}' | grep -c "True")" -eq 1 ];
+  do
+    echo "Approving csrs ..."
+    oc get csr -o go-template='{{range .items}}{{if not .status}}{{.metadata.name}}{{"\n"}}{{end}}{{end}}' | xargs --no-run-if-empty oc adm certificate approve &> /dev/null || true
+    sleep 30
+  done
+}
+
+function wait_for_cvo {
+  echo "Waiting for cvo"
+  until [ "$(oc get clusterversion -o jsonpath='{.items[0].status.conditions[?(@.type=="Available")].status}')" == "True" ];
```

Comment:
What is the motivation for waiting on the cluster-version operator declaring the install complete?  On typical installs, we lose the ability to run `openshift-install gather bootstrap ...` after bootstrap-teardown.  This post-reboot script is the bootstrap-in-place analog of post-bootstrap-teardown, right?  Can't we rely on must-gather and other day-2 tooling at that point, without having to preserve assets to support `openshift-install gather bootstrap ...`?

Reply:
On typical installs, the bootstrap goes down only once we have a working control plane on the master nodes.

We can replace `wait_for_cvo` with waiting for the same `required-pods` cluster-bootstrap waits for:
"openshift-kube-apiserver/kube-apiserver,openshift-kube-scheduler/openshift-kube-scheduler,openshift-kube-controller-manager/kube-controller-manager,openshift-cluster-version/cluster-version-operator"

Yet I don't think there is much value in deleting the bootstrap logs early.

Reply:
waiting for the same required-pods would further reduce the differences in this workflow.

Reply:
I agree it will reduce the differences., but I'm not sure it will work.
In a regular installation, etcd will keep running and the static pods resources and secrets will still exist under /etc/kubernetes.
Also, I think that if we really want it we should add support for this wait option in cluster-bootstrap. meaning it should just wait without setting up a temporary control plane. 
Trying to replicate the cluster-bootstrap code in bash seems wrong.  
 So I suggest keeping the current implementation and have another task for checking if it's possible to do the cleanup earlier, if it's possible we can add the option to cluster-bootstrap and use it.